MSAL/src/configuration/external/ios/MSALLegacySharedAccountsProvider.m

```diff
@@ -436,6 +451,12 @@ - (BOOL)saveUpdatedAccount:(id<MSALAccount>)account
     
     for (MSALLegacySharedAccount *sharedAccount in accounts)
     {
+        if (operation == MSALLegacySharedAccountWipeOperation)
+        {
+            resultDictionary[sharedAccount.accountIdentifier] = nil;
```

Comment:
Should we check for the existence of key in the resultDictionary before assigning it to nil, something like msidSetObjectIfNotNil.

Reply:
Thanks for the suggestion. Would it make any difference in terms of performance?

Reply:
I think above code will crash if the resultDictionary does not have the key "sharedAccount.accountIdentifier" so we might need to check for the existence of the key. :)

Reply:
It won't crash if the object is nil: https://developer.apple.com/documentation/foundation/nsmutabledictionary/1574187-setobject

If key is nil, it will, but we check for key presence when creating the account. I will add an additional check for key for this particular method just for future safety, but I don't think we need to check for the value. Is that what you had in mind?

Reply:
I feel that in this method, there is a possibility that resultDictionary is [NSMutableDictionary new] and then we set resultDictionary[sharedAccount.accountIdentifier] = nil , I mean that this code will crash if resultDictionary does not have the key "sharedAccount.accountIdentifier".  sharedAccount.accountIdentifier can or cannot have a value but it still needs to be present in the dictionary right. Sorry if my question is not clear. We can discuss tomorrow.

Reply:
For instance, this code crashes
`        NSMutableDictionary *dict = [NSMutableDictionary new];
        [dict setObject:@"rohit" forKey:nil];`
with exception "Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSDictionaryM setObject:forKey:]: key cannot be nil"

Reply:
yes, there's a difference between:
[dict setObject:nil forKey:@"rohit"]; expected to crash
vs 
dict[@"rohit"] = nil; - won't crash as per this document: https://developer.apple.com/documentation/foundation/nsmutabledictionary/1574187-setobject:

```
object
The value for aKey. A strong reference to the object is maintained by the dictionary.
Passing nil will cause any object corresponding to aKey to be removed from the dictionary.
```

Reply:
I think what the above documentation means is that if you set nil for the object value for an existing key in the dictionary, it will remove the key from the dictionary so it is correct/expected. But the key still needs to be present first in the dictionary before assigning its corresponding object to nil. If you try to run this code in xcode, it will crash.

Reply:
Ok ignore my comment. :) I see what you mean, there is a difference in behavior between these two methods 
dict[@"rohit"] = nil; 
[dict setObject:nil forKey:@"rohit"];